sdk/src/main/java/com/hedera/hashgraph/sdk/Executable.java

```diff
@@ -156,12 +162,72 @@ public SdkRequestT setNodeAccountIds(List<AccountId> nodeAccountIds) {
         if (minBackoff == null) {
             minBackoff = client.getMinBackoff();
         }
+    }
 
-        return onExecuteAsync(client).thenCompose((v) -> {
-            if (nodeAccountIds.isEmpty()) {
-                throw new IllegalStateException("Request node account IDs were not set before executing");
+    void delay(long delay) {
+        try {
+            Thread.sleep(delay);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public O execute(Client client) throws TimeoutException, PrecheckStatusException {
+        return execute(client, client.getRequestTimeout());
+    }
+
+    public O execute(Client client, Duration timeout) throws TimeoutException, PrecheckStatusException {
+        Throwable lastException = null;
+
+        mergeFromClient(client);
+        onExecute(client);
+        checkNodeAccountIds();
+        setNodesFromNodeAccountIds(client);
+
+        for (int attempt = 1; attempt <= maxAttempts + 1; attempt++) {
```

Comment:
This is doing one more attempt than it does for async. It's always been a bit confusing if this property is tracking "retry attempts" or "total attempts", with the initial attempt only being counted in the latter. But traditionally the async has started at one since it tracked total attempts.